addons/hr_expense/models/hr_expense.py

```diff
@@ -0,0 +1,320 @@
+            raise UserError(_("You can only submit draft expenses!"))
+        self.write({'state': 'submit'})
+
+    @api.multi
+    def approve_expenses(self):
+        self.write({'state': 'approve'})
+
+    @api.multi
+    def refuse_expenses(self, reason):
+        self.write({'reason': reason, 'state': 'cancel'})
+
+    @api.multi
+    def paid_expenses(self):
+        self.write({'state': 'done'})
+
+    @api.multi
+    def reset_expenses(self):
+        return self.write({'state': 'draft'})
+
+    @api.multi
+    def _track_subtype(self, init_values):
+        self.ensure_one()
+        if 'state' in init_values and self.state == 'approve':
+            return 'hr_expense.mt_expense_approved'
+        elif 'state' in init_values and self.state == 'submit':
+            return 'hr_expense.mt_expense_confirmed'
+        elif 'state' in init_values and self.state == 'cancel':
+            return 'hr_expense.mt_expense_refused'
+        return super(HrExpense, self)._track_subtype(init_values)
+
+    def _line_get_convert(self, x, part, date):
+        partner_id  = self.pool.get('res.partner')._find_accounting_partner(part).id
+        return {
+            'date_maturity': x.get('date_maturity', False),
+            'partner_id': partner_id,
+            'name': x['name'][:64],
+            'date': date,
+            'debit': x['price']>0 and x['price'],
+            'credit': x['price']<0 and -x['price'],
+            'account_id': x['account_id'],
+            'analytic_line_ids': x.get('analytic_line_ids', False),
+            'amount_currency': x['price']>0 and abs(x.get('amount_currency', False)) or -abs(x.get('amount_currency', False)),
+            'currency_id': x.get('currency_id', False),
+            'tax_line_id': x.get('tax_line_id', False),
+            'ref': x.get('ref', False),
+            'quantity': x.get('quantity',1.00),
+            'product_id': x.get('product_id', False),
+            'product_uom_id': x.get('uos_id', False),
+            'analytic_account_id': x.get('account_analytic_id', False),
+        }
+
+    @api.multi
+    def _compute_expense_totals(self, company_currency, account_move_lines):
+        '''
+        internal method used for computation of total amount of an expense in the company currency and
+        in the expense currency, given the account_move_lines that will be created. It also do some small
+        transformations at these account_move_lines (for multi-currency purposes)
+        
+        :param account_move_lines: list of dict
+        :rtype: tuple of 3 elements (a, b ,c)
+            a: total in company currency
+            b: total in hr.expense currency
+            c: account_move_lines potentially modified
+        '''
+        total = 0.0
+        total_currency = 0.0
+        for line in account_move_lines:
+            line['currency_id'] = False
+            line['amount_currency'] = False
+            if self.currency_id != company_currency:
```

Comment:
If decorator is @api.multi and if it's going to use only one record(like, self.currency_id) then you would need to add self.ensure_one() at the beginning of the method